module/VuFindConsole/tests/unit-tests/src/VuFindTest/Command/Util/IndexReservesCommandTest.php

```diff
@@ -330,9 +350,260 @@ public function testSuccessWithILS()
         $commandTester = new CommandTester($command);
         $commandTester->execute([]);
         $this->assertEquals(0, $commandTester->getStatusCode());
-        $this->assertEquals(
-            "Successfully loaded 3 rows.\n",
+        $this->assertStringContainsString(
+            'Successfully loaded 3 rows.',
+            $commandTester->getDisplay()
+        );
+    }
+
+    /**
+     * Test successful ILS loading with some invalid data in reserves.
+     *
+     * @return void
+     */
+    public function testSuccessWithILSWithInvalidData()
+    {
+        $ils = $this->getMockIlsConnection();
+        $instructors = ['inst1' => 'inst1'];
+        $courses = ['course1' => 'course1'];
+        $departments = ['dept1' => 'dept1'];
+        $reserves = [
+            [
+                'BIB_ID' => 1,
+                'COURSE_ID' => 'course2',
+                'DEPARTMENT_ID' => 'dept2',
+                'INSTRUCTOR_ID' => 'inst2',
+            ],
+        ];
+        $this->expectConsecutiveCalls(
+            $ils,
+            '__call',
+            [
+                ['getInstructors'],
+                ['getCourses'],
+                ['getDepartments'],
+                ['findReserves'],
+            ],
+            [
+                $instructors,
+                $courses,
+                $departments,
+                $reserves,
+            ]
+        );
+
+        $writer = $this->getMockSolrWriter();
+        $that = $this;
```

Comment:
In PHP anonymous functions, you don't need to use `$that` -- you can just call `$this` for access to the object containing the function. I think you can simplify this code as a result!

(And if you copy-pasted this from somewhere else, that's my fault -- I only learned this relatively recently).

Reply:
I seem to get `Cannot use $this as lexical variable` when trying to directly use `$this`. It throws the error on this line: `$updateValidator = function ($update) use ($this) {`

Reply:
Disregard -- I had an "oh duh" moment and realized it shouldn't even be passed in the `use ($this)`. Works fine just using `$this` within it.

Reply:
Sorry, should have mentioned that detail and saved you a moment's confusion! :-)